airbyte-cdk/python/airbyte_cdk/sources/streams/http/requests_native_auth/abstract_oauth.py

```diff
@@ -29,6 +31,20 @@ class AbstractOauth2Authenticator(AuthBase):
 
     _NO_STREAM_NAME = None
 
+    def __init__(
+        self,
+        refresh_token_error_status_codes: Tuple[int, ...] = (),
+        refresh_token_error_key: str = "",
+        refresh_token_error_values: Tuple[str, ...] = (),
+    ) -> None:
+        """
+        If all of refresh_token_error_status_codes, refresh_token_error_key, and refresh_token_error_values are set,
+        then http errors with such params will be wrapped in AirbyteTracedException.
+        """
+        self._refresh_token_error_status_codes = refresh_token_error_status_codes
```

Comment:
nit: is there value in setting only of these fields? If not, I would recommend wrapping them in a struct to ensure they are either all set, or none of them are

```
@dataclass
class RefreshTokenErrorConfig:
    refresh_token_error_status_codes: Tuple[int, ...]
    refresh_token_error_key: str
    refresh_token_error_values: Tuple

NO_ERROR_CONFIG = RefreshTokenErrorConfig((), "", ())

class AbstractOauth2Authenticator(AuthBase):
    def __init__(
        self,
        ...,
        refesh_token_error_config: RefreshTokenErrorConfig =    NO_ERROR_CONFIG)
```